controlsfx/src/main/java/org/controlsfx/tools/Utils.java

```diff
@@ -44,14 +44,16 @@
      */
     public static Window getWindow(Object owner) throws IllegalArgumentException {
         if (owner == null) {
+            Window window = null;
             // lets just get the focused stage and show the dialog in there
-            List<Window> windows = Window.getWindows();
-            for (Window window : windows) {
+            Iterator<Window> windows = Window.getWindows().iterator();
+            while (windows.hasNext()) {
+                window = windows.next();
                 if (window.isFocused() && !(window instanceof PopupWindow)) {
-                    return window;
+                    break;
                 }
             }
-            return null;
+            return window;

```

Comment:
In case no stage is shown, null would still be returned and the NPE would still occurs if I'm not mistaken ?